ext/DiffEqBaseEnzymeExt.jl

```diff
@@ -0,0 +1,53 @@
+module DiffEqBaseEnzymeExt
+
+using DiffEqBase
+import DiffEqBase: value
+isdefined(Base, :get_extension) ? (import Enzyme) : (import ..Enzyme)
+
+using ChainRulesCore
+using EnzymeCore
+
+function EnzymeCore.EnzymeRules.augmented_primal(config::EnzymeCore.EnzymeRules.ConfigWidth{1}, func::Const{typeof(DiffEqBase.solve_up)}, ::Type{Duplicated{RT}}, prob, sensealg, u0, p, args...; kwargs...) where RT
```

Comment:
```suggestion
function EnzymeCore.EnzymeRules.augmented_primal(config::EnzymeCore.EnzymeRules.ConfigWidth{1}, func::Const{typeof(DiffEqBase.solve_up)}, ::Type{Duplicated{RT}}, prob, sensealg::Union{Const{Nothing}, Const{<:AbstractSensitivityAlgorithm}}, u0, p, args...; kwargs...) where RT
```

this is important because the adjoint method specifically wants to pass through if `sensealg` is something different. This allows us to implement `sensealg=EnzymeAdjoint()` which works (or is supposed to work, will need to check) by differentiating the solver itself, rather than using the adjoint, which has different numerical and performance properties.

Reply:
that's fine, just also make sure to add it to the reverse as well!